src/report-table/report-table.component.tsx

```diff
@@ -52,7 +62,7 @@ const ReportTable = ({patientUuid}) => {
             day: '2-digit',
           }),
           requester: '-',
-          file: `${row.resource.presentedForm[0].title}.pdf`,
+          file: fileName,
```

Comment:
you can extract the responsibility to derive filename into its own isolation

```
function deriveFileName(title, contentType) {
	if (contentType.endsWith('pdf'))
		return `${title}.pdf`
  else if (contentType.endsWith('jpg'))
    return `${title}.jpg`
  else if (contentType.endsWith('jpeg'))
    return `${title}.jpeg`
  return ''
}

```
then you could just attach it to the object without the need of any garbage variables or mutations
```
const title = row.resource.presentedForm[0].title
const contentType = row.resource.presentedForm[0].contentType
...
...
          requester: '-',
          file: deriveFileName(title, contentType),
```

Reply:
Changes have been made according to the inputs above.